server/src/main/java/org/elasticsearch/rest/BaseRestHandler.java

```diff
@@ -81,10 +80,7 @@ public final void handleRequest(RestRequest request, RestChannel channel, NodeCl
 
         // validate unconsumed params, but we must exclude params used to format the response
         // use a sorted set so the unconsumed parameters appear in a reliable sorted order
-        final SortedSet<String> unconsumedParams = request.unconsumedParams()
-            .stream()
-            .filter(p -> responseParams(request.getRestApiVersion()).contains(p) == false)
-            .collect(Collectors.toCollection(TreeSet::new));
+        final SortedSet<String> unconsumedParams = new TreeSet<>(request.unconsumedParams());
```

Comment:
I approve of this change :+1: but it does mean that we have to validate all response parameters up front. This PR does not do that, so e.g. `./gradlew :rest-api-spec:yamlRestTest` fails rather catastrophically right now.